### PR TITLE
feat: initial refactor

### PR DESCRIPTION
**Pull request by Plork**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,4 +12,3 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 #### 🐛 Bug Fixes
```
Comment (marwinbaumannsbp) [inline CHANGELOG.md:12]: This v0.6.0 should not be here

File: examples/basic/versions.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.3"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.39.0"
```
Comment (marwinbaumannsbp) [inline examples/basic/versions.tf:7]: This versions.tf file is not the same as the versions.tf of the main module

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,110 @@
+variable "cluster_arn" {
+  type        = string
+  description = "The ARN of the ECS cluster"
+  default     = null
+}
+
+variable "config" {
+  description = "A map containing all labeler configuration options"
```
Comment (marwinbaumannsbp) [inline variables.tf:8]: are these variables described somewhere? Can we link to the energylabeler documentation? For some its hard to know what can be entered.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  type = object({
```
Comment (marwinbaumannsbp) [inline variables.tf:9]: Are these all optional?

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,110 @@
+variable "cluster_arn" {
```
Comment (marwinbaumannsbp) [inline variables.tf:1]: could you make a choice in orderint e.g. description - type - default - validation and make thsi consistent accross the variables

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  validation {
+    condition     = alltrue([for o in var.security_group_egress_rules : (o.cidr_ipv4 != null || o.cidr_ipv6 != null || o.prefix_list_id != null || o.referenced_security_group_id != null)])
+    error_message = "Although \"cidr_ipv4\", \"cidr_ipv6\", \"prefix_list_id\", and \"referenced_security_group_id\" are all marked as optional, you must provide one of them in order to configure the destination of the traffic."
+  }
+}
+
+variable "subnet_ids" {
+  type        = list(string)
+  default     = null
+  description = "The subnet ids where this lambda needs to run"
```
Comment (marwinbaumannsbp) [inline variables.tf:103]: the description mentions lambda and also its unclear to me what these subnet_ids are used for, should these not be mandatory or can we run this solution without a vpc?

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    report-suppressed-findings  = optional(bool, false)
+    account-thresholds          = optional(string)
+    zone-thresholds             = optional(string)
+    security-hub-query-filter   = optional(string)
+    validate-metadata-file      = optional(string)
+  })
+  default = {}
+  validation {
+    condition     = length(compact([var.config.single-account-id, var.config.audit-zone-name, var.config.organizations-zone-name])) == 1
+    error_message = "Parameters organizations-zone-name, audit-zone-name and single-account-id are mutually exclusive"
```
Comment (marwinbaumannsbp) [inline variables.tf:33]: what is audit-zone-name?

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+  id = var.subnet_ids[0]
+}
+
+resource "aws_security_group" "default" {
+  #checkov:skip=CKV2_AWS_5: False positive finding, the security group is attached.
+  count = var.subnet_ids != null ? 1 : 0
+
+  name        = local.name
+  description = "Security group for ESC ${local.name}"
```
Comment (marwinbaumannsbp) [inline main.tf:55]:
```suggestion
  description = "Security group for ECS ${local.name}"
```

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  prefix_list_id               = each.value.prefix_list_id
+  referenced_security_group_id = each.value.referenced_security_group_id
+  security_group_id            = aws_security_group.default[0].id
+  to_port                      = each.value.to_port
+}
+
+resource "aws_ecs_cluster" "default" {
+  count = var.cluster_arn == null ? 1 : 0
+
+  name = "ecs-scheduled-task"
```
Comment (marwinbaumannsbp) [inline main.tf:81]: should there not be some "energylabeler" or somethign in the name?

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+  }
+}
+
+module "task_role" {
+  source  = "schubergphilis/mcaf-role/aws"
+  version = "~> 0.4.0"
+
+  name                  = local.ecs_task_iam_name
+  create_policy         = true
+  permissions_boundary  = null
```
Comment (marwinbaumannsbp) [inline main.tf:142]: can you create a variable to pass down the permissions_boundary arn?

Comment (marwinbaumannsbp) [reply main.tf:142]: same for the other roles, maybe 1 var to rule them all

File: main.tf [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+    }
+  }
+}
+
+module "aws_ecs_container_definition" {
+
+  source  = "terraform-aws-modules/ecs/aws//modules/container-definition"
+  version = "~> 5.11.4"
+
+  cloudwatch_log_group_name       = "/aws/ecs/${local.name}"
```
Comment (marwinbaumannsbp) [inline main.tf:205]: can you make a var out of `/aws/ecs/` and default them to `/aws/ecs/` some customers need a custom prefix

File: variables.tf (hunk 3 above)
Comment (Plork) [reply variables.tf:8]: it's even hard to find with the documentation.  I have to go into the lib from the source of the cli to find any of it :S

asked the maintainers to update the documentation but can provide a link to the readme

File: variables.tf (hunk 4 above)
Comment (Plork) [reply variables.tf:9]: I could not find it in the cli what is and what isn't

File: variables.tf (hunk 7 above)
Comment (Plork) [reply variables.tf:33]: /shrug ... it's passed to the cli

File: main.tf (hunk 9 above)
Comment (Plork) [reply main.tf:81]: I don't think the task and the cluster should be or are related

File: main.tf (hunk 10 above)
Comment (Plork) [reply main.tf:142]: sure

File: variables.tf (hunk 7 above)
Comment (Plork) [reply variables.tf:33]: https://awsenergylabelerlib.readthedocs.io/en/latest/usage.html ?

Comment (Plork) [reply variables.tf:33]: https://aws-energy-labeler-cli.readthedocs.io/en/latest/usage.html

File: main.tf (hunk 9 above)
Comment (marwinbaumannsbp) [reply main.tf:81]: That makes sense but if one opts for letting this moduel create the cluster its not going to be re-used by other thigns right? That would not make sense to me tbh

File: variables.tf (hunk 6 above)
Comment (Plork) [reply variables.tf:103]: in theory we could, unless you want to store the results on an S3. So it depends?

File: variables.tf (hunk 3 above)
Comment (Plork) [reply variables.tf]: https://github.com/schubergphilis/terraform-aws-mcaf-energy-labeler/blob/13452ff07183a176e80e2dccdc58f7182ecb866f/README.md#terraform-aws-mcaf-energy-labeler

first documentation links to the library it's based on.

File: main.tf (hunk 9 above)
Comment (Plork) [reply main.tf]: default name is now used

File: main.tf (hunk 11 above)
Comment (Plork) [reply main.tf]: consistent with all the other mcaf module atm. So leaving this for a future Pr to fix if someone needs it.

**Review by shoekstra — APPROVED** [review]

**Pull request merged**